paasta_tools/kubernetes_tools.py

```diff
@@ -1163,12 +1163,16 @@ def get_all_pods(kube_client: KubeClient, namespace: str = "paasta") -> Sequence
 def filter_pods_by_service_instance(
     pod_list: Sequence[V1Pod], service: str, instance: str
 ) -> Sequence[V1Pod]:
-    return [
-        pod
-        for pod in pod_list
-        if pod.metadata.labels["yelp.com/paasta_service"] == service
-        and pod.metadata.labels["yelp.com/paasta_instance"] == instance
-    ]
+    pod_search_results = []
+    for pod in pod_list:
```

Comment:
can we retain list comprehension form of this function? you can add `pod.metadata.labels is not None and ...` to the `if` condition.

Reply:
We can. Added. 👍